### PR TITLE
When receiving faulty JSON-RPC on the socket, send an error back to the caller

### PR DESCRIPTION
**Pull request by blinry**
This should make it easier to develop editor plugins, when you get feedback that your message couldn't get parsed.

Partial fix for #29.

### CODE REVIEW

File: daemon/src/types.rs [hunk 1]
```diff
@@ -139,8 +139,8 @@ pub enum JSONRPCFromEditor {
 }
 impl JSONRPCFromEditor {
     pub fn from_jsonrpc(jsonrpc: &str) -> Result<Self, anyhow::Error> {
-        let error_message = format!("Failed to deserialize editor message: {jsonrpc}");
-        let message = serde_json::from_str(jsonrpc).expect(&error_message);
+        //let error_message = format!("Failed to deserialize editor message: {jsonrpc}");
+        let message = serde_json::from_str(jsonrpc)?; //.context(error_message)?;
```
Comment (zormit) [inline daemon/src/types.rs]: Should we remove the comments here, aka, are you happy with the "question mark only"? Or was this something not yet working?

**Review by zormit — APPROVED** [review]
looks good to me, just left one comment on the comment :)

(yay, an `expect` less! =))

File: daemon/src/types.rs (hunk 1 above)
Comment (blinry) [reply daemon/src/types.rs]: Ah no, it was an oversight. I removed the comments.

**Comment by blinry** [comment]
Ideas for how to debug the failing fuzzer further:

- [ ] Enable debugging in the fuzzer, and look hard at the output?
- [ ] Check if it's always the case that daemon == nvim and peer == nvim2 (is it a peering problem?)
- [ ] Change the fuzzer so that it runs relatively fast locally (decrease sleeps), but increase number of steps, and with that configuration, bisect and see where it starts failing

**Comment by zormit** [comment]
I think this one can be merged as well now :)

**Pull request merged**